src/analysis/mean_inequalities.lean

```diff
@@ -125,6 +125,43 @@ begin
       { rw [exp_log hz] } } }
 end
 
+theorem geom_mean_weighted_of_constant (w z : ι → ℝ) (x : ℝ) (hw : ∀ i ∈ s, 0 ≤ w i)
+  (hw' : ∑ i in s, w i = 1) (hz : ∀ i ∈ s, 0 ≤ z i) (hx : ∀ i ∈ s, w i ≠ 0 → z i = x) :
```

Comment:
Is it really more natural here to assume `hz` than the plain `0 ≤ x`?

Reply:
`hz` is the right assumption for the future iff version, but the derivation of `0 ≤ x` can happen in that proof too. I don't care much either way; let me know if you feel I should change it.

Reply:
No, it's ok, let's keep it like that.